src/zope/publisher/browser.py

```diff
@@ -813,6 +814,8 @@ def reset(self):
 
 def isHTML(str):
      """Try to determine whether str is HTML or not."""
+     if isinstance(str, six.binary_type):
+         str = str.decode()
```

Comment:
What happens if this raises a UnicodeDecodeError?

Reply:
Nothing good, probably. :) I've added a `catch` now. Thank you for your dilligent review!